### PR TITLE
chore: migration from google place api to mapbox

### LINKED ISSUE #1969: Replace google places API with Mapbox API
**Describe the feature you'd like**
Documentation: https://docs.mapbox.com/android/plugins/overview/places/

**Would you like to work on the issue?**

Open for all

### PR DESCRIPTION
**Pull request by naman653**
Fixes #1969

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4479632&bid=15117536)

File: app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 1]
```diff
@@ -304,38 +297,49 @@ public void close() {
...
-            @Override
-            public void onPlaceSelected(Place place) {
-
-                Timber.d(place.getAddress());
-                Event event = binding.getEvent();
-                event.latitude = place.getLatLng().latitude;
-                event.longitude = place.getLatLng().longitude;
-                event.locationName = place.getAddress();
-                event.searchableLocationName = place.getName();
+            if (mapboxAccessToken.equals("YOUR_ACCESS_TOKEN")) {
```
Comment (iamareebjamal) [inline app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4479632)

File: app/src/playStore/java/com/eventyay/organizer/core/event/create/EventDetailsStepOne.java [hunk 2]
```diff
@@ -92,31 +91,41 @@ private void setupPlacesAutocomplete() {
...
-            @Override
-            public void onPlaceSelected(Place place) {
-
-                Timber.d(place.getAddress());
-                Event event = binding.getEvent();
-                event.latitude = place.getLatLng().latitude;
-                event.longitude = place.getLatLng().longitude;
-                event.locationName = place.getAddress();
-                event.searchableLocationName = place.getName();
+            if (mapboxAccessToken.equals("YOUR_ACCESS_TOKEN")) {
```
Comment (iamareebjamal) [inline app/src/playStore/java/com/eventyay/organizer/core/event/create/EventDetailsStepOne.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4479632)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4479632&bid=15117536)

File: app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 3]
```diff
@@ -304,38 +297,49 @@ public void close() {
...
-            @Override
-            public void onPlaceSelected(Place place) {
-
-                Timber.d(place.getAddress());
-                Event event = binding.getEvent();
-                event.latitude = place.getLatLng().latitude;
-                event.longitude = place.getLatLng().longitude;
-                event.locationName = place.getAddress();
-                event.searchableLocationName = place.getName();
+            if (mapboxAccessToken.equals(getString(R.string.your_access_token))) {
```
Comment (iamareebjamal) [inline app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4479632)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4479632&bid=15117536)

File: app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 4]
```diff
@@ -304,38 +297,49 @@ public void close() {
...
-            @Override
-            public void onPlaceSelected(Place place) {
-
-                Timber.d(place.getAddress());
-                Event event = binding.getEvent();
-                event.latitude = place.getLatLng().latitude;
-                event.longitude = place.getLatLng().longitude;
-                event.locationName = place.getAddress();
-                event.searchableLocationName = place.getName();
+            if (getString(R.string.your_access_token).equals(mapboxAccessToken)) {
```
Comment (iamareebjamal) [inline app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4479632)

Comment (naman653) [reply app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: Why only this line is getting an error. Other files also contain the same line but they are passing the codacy check just fine.

Comment (iamareebjamal) [reply app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: This is a warning, not an error

Comment (naman653) [reply app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: But why other files are not receiving the warning?

**Comment by naman653** [comment]
@liveHarshit Please review.

**Comment by iamareebjamal** [comment]
Now, what's the difference in fdroid and play store variant in Event Wizard?

If there is no difference, why not merge it

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java  7
- app/src/main/java/com/eventyay/organizer/core/event/create/EventDetailsStepOne.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-app/open-event-orga-app/pullRequest?prid=4479632&bid=15117536)

**Comment by ShridharGoel** [comment]
@naman653 Any updates on this?

**Comment by ShridharGoel** [comment]
Done in #1986

**Pull request closed without merging**